src/cargo/core/shell.rs

```diff
@@ -565,6 +576,14 @@ fn supports_unicode(stream: &dyn IsTerminal) -> bool {
     !stream.is_terminal() || supports_unicode::supports_unicode()
 }
 
+#[allow(clippy::disallowed_methods)] // ALLOWED: to read terminal app signature
+fn supports_progress_report() -> bool {
```

Comment:
This should be dependent on the state of the stream.

If stderr is being piped to a file, we shouldn't emit these ANSI escape codes (unless the user overrides t)

Reply:
Out of curiosity, who is going to maintain this detection logic as the terminal support for this feature increases?

Reply:
For now, us. As I said earlier in this PR, I expect this will get split out into a crate.

Reply:
> If stderr is being piped to a file, we shouldn't emit these ANSI escape codes (unless the user overrides t)

Makes sense. I've modified the condition to `*progress_report && *stderr_tty`